tools/bridge/bridge/event_fetcher.py

```diff
@@ -54,14 +59,24 @@ def __init__(
             before_sleep=tenacity.before_sleep_log(self.logger, logging.WARN),
         )
 
-    def _rpc_get_node_status(self):
-        return self._retrying.call(node_status.get_node_status, self.web3)
+    def _rpc_get_cached_node_status(self):
+        if (
+            node_status is None
+            or time.time() - self._node_status_timestamp
+            > NODE_STATUS_CACHE_TIME_SECONDS
+        ):
+            # To reduce number of node calls we cache the result
+            self._node_status = self._retrying.call(
+                node_status.get_node_status, self.web3
+            )
+            self._node_status_timestamp = time.time()
```

Comment:
I would probably make the timestamp a field in `node_status.NodeStatus`. But we can also merge as is.